src/ui/manage-ui/src/FamilyHubs.ServiceDirectory.Admin.Web/Pages/manage-services/What-Language.cshtml.cs

```diff
@@ -71,26 +71,27 @@ protected override void OnGetWithError()
     protected override void OnGetWithModel()
     {
         // we've redirected to self with user input and no errors, so javascript must be disabled
-        if (ServiceModel?.UserInput != null)
+        
+        // redirectingToSelf is only set when adding a new field
+        if(ServiceModel?.UserInput != null && Request.Query.TryGetValue("redirectingToSelf", out var redirectToSelf) && redirectToSelf == "true")
         {
             SetFormData();
+            return;
         }
-
+        
         // default to no language selected
         UserLanguageOptions = LanguageOptions.Take(1);
-
         if (ServiceModel!.LanguageCodes?.Any() == true)
         {
-            UserLanguageOptions = ServiceModel!.LanguageCodes.Select(l =>
+            UserLanguageOptions = ServiceModel!.LanguageCodes.Select(lang =>
             {
                 //todo: put into method
```

Comment:
Might be worth actioning this TODO and putting lines 83 to 92 into a little private method

Reply:
I've moved the code out into seperate methods now